### ISSUE TITLE
Typescript don't detect type error in implicitly typed arrow function

### ISSUE THREAD

**Issue by Stas-tolpekin** [microsoft/TypeScript#46535]
# Bug Report
### 🔎 Search Terms

arrow function expression type return type

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ

### ⏯ Playground Link
https://www.typescriptlang.org/play?ts=4.5.0-beta#code/FASwdgLgpgTgZgQwMZQAQFkD2ATKAbVAb2FVNThH2wEYAuVMAVwFsAjWAbhLIqoCZ6AZwgxwAcy4BfYMAD0s1NQB0qEMwAOeEEhARUEAJ7qo2VAhgxMAd3KMwSCCExgVAeTBpMcciDwnBqF6oAEpQEIwwYIGsAFZQDqoBWLh4AAYBvHjYwEjOwuRgdKgAFMw4+PTJ+ACUqAC8AHyoAArmjgh4ADxVeE11JWUptY0lxGQ+VADM9NR8kwA03KSZNDOLktUy8qh8KlAAHprauvpGJmYW1rb2js45eXpwYAID5XiVb8NNrTDtXT19V4pD5Deg-P7dN59JrFMY8ShZaaKOaLcYrIrUdabOQKSYqNRHHR6QzGUzmSw2OB2BxOFwhMIRKKYWLxPTYTBQAJgTB6XKQBDgMxgAwYN7pCZZQT3MD5J5I0pvEE1erfNogDqQlKAhVDFWjJYS7BI2YLYAbIA
### 💻 Code
```typescript
interface Model {
    field1: number;
    field2: string;
}

// Implicit typed arrow function.
// One of fileds of Return object is Model`s field.
// Another field is not part of interface and should be mark as error.
const fn1: (model: Model) => Partial<Model> = (model) => ({
    field3: 123,
    field1: 1,
})
```

if I remove using of field1 in this statement TypeScript will mark field3 as the wrong key. (more examples in Playground)

### 🙁 Actual behavior

The field that is not part of the Model interface is not marked as an error.

### 🙂 Expected behavior

The field that is not part of the Model interface should be marked as an error
